components/leaderboard/RankingsTable.tsx

```diff
@@ -66,63 +80,78 @@ const RankingsTable: FunctionComponent<RankingProps> = ({
-                    style={{
-                      color:
-                        selectedAddress === item.address
-                          ? "#6AFFAF"
-                          : "#ffffff",
-                    }}
-                  >
-                    {isMobile &&
-                    item &&
-                    item.displayName &&
-                    isStarkDomain(item.displayName)
-                      ? minifyDomain(item.displayName)
-                      : item.displayName}
-                  </p>
-                </div>
-              </div>
-              <div className={styles.ranking_table_row_xp_quest}>
-                <div className={styles.ranking_points_layout}>
-                  <CDNImage
-                    src={"/icons/xpBadge.svg"}
-                    priority
-                    width={35}
-                    height={35}
-                    alt="xp badge"
-                  />
-                  <p className="text-white text-center">{item.xp}</p>
+        <>
+          {isTop50RankedView && (
+            <Top3RankedUsers leaderboardToppers={leaderboardToppers} />
+          )}
+          {slicedDisplayedData?.map((item, index) => {
+            const isMyRank =
+              index === 0 && item.address === connectedUserAddress; // According to api, first item should be connected user rank
+            const itemRank =
+              data.first_elt_position + index + (isTop50RankedView ? 3 : 0);
+            return (
+              <Link key={item.address} href={`/${decimalToHex(item.address)}`}>
+                <div
+                  className={styles.ranking_table_row}
+                  style={{
+                    backgroundColor:
+                      selectedAddress === item.address || isMyRank
+                        ? "black"
+                        : hoveredIndex === index
+                        ? "#1C1C1C"
+                        : "transparent",
+                  }}
+                  onMouseOver={() => setHoveredIndex(index)}
+                >
+                  <div className={styles.ranking_table_row_name_rank}>
+                    <div className={styles.ranking_position_layout}>
+                      <p className="text-white text-center">
+                        {itemRank > 10000
+                          ? "+10k"
+                          : itemRank > 5000
+                          ? "+5k"
+                          : addNumberPadding(itemRank)}
+                      </p>
+                    </div>
+                    <div className={styles.ranking_profile_layout}>
+                      <Avatar address={item.address} width="32" />
+                      <p
+                        style={{
+                          color:
+                            selectedAddress === item.address || isMyRank
+                              ? "#6AFFAF"
+                              : "#ffffff",
+                        }}
+                      >
+                        {isMobile &&
```

Comment:
useMemo this value and avoid inline longer conditions which have state variables involved. React will recompute them even if they haven't changed